### PR TITLE
Add support for JSON and YAML chain and token configs

### PR DESCRIPTION
**Pull request by jmrossy**
Fixes https://github.com/hyperlane-xyz/issues/issues/732

- Add support for JSON and YAML chain and token configs
- Update some stale docs
- Cleanup default chain and token const files

### CODE REVIEW

**Comment by avious00** [comment]
thanks for adding the additional comments at the top of the snippets in `/chains.yaml` + ` /chains.ts`

File: src/consts/tokens.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# A list of warp token configs
```
Comment (nambrot) [inline src/consts/tokens.yaml]: I think we need to get a bit crisper IMO with our language. Very easy to add the warp token config that is input to the warp deploy command

File: src/features/chains/metadata.ts [hunk 2]
```diff
@@ -12,9 +12,13 @@ import {
 } from '@hyperlane-xyz/sdk';
 import { ProtocolType } from '@hyperlane-xyz/utils';
 
-import { chains } from '../../consts/chains';
+import { chains as ChainsTS } from '../../consts/chains';
+import ChainsJson from '../../consts/chains.json';
```
Comment (nambrot) [inline src/features/chains/metadata.ts:16]: Do we even have a thing that outputs json chain metadata?

File: src/features/tokens/metadata.ts [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import { EvmTokenAdapter, ITokenAdapter, TokenType } from '@hyperlane-xyz/sdk';
 import { ProtocolType } from '@hyperlane-xyz/utils';
 
-import { tokenList } from '../../consts/tokens';
+import { tokenList as TokensTS } from '../../consts/tokens';
+import TokensJson from '../../consts/tokens.json';
+import TokensYaml from '../../consts/tokens.yaml';
```
Comment (nambrot) [inline src/features/tokens/metadata.ts:6]: For warp token artifacts, they are not currently being output with tokens.yaml as the file name right?

**Review by nambrot — APPROVED** [review]
🙏 for fixing this

We are due for a more holistic overhaul of artifacts that are reused across, but one step at a time :)

File: src/features/chains/metadata.ts (hunk 2 above)
Comment (jmrossy) [reply src/features/chains/metadata.ts:16]: The CLI commands can output either. There's a `format` flag that can be json or yaml.

File: src/consts/tokens.yaml (hunk 1 above)
Comment (jmrossy) [reply src/consts/tokens.yaml]: Good point, I'll rephrase

File: src/features/tokens/metadata.ts (hunk 3 above)
Comment (jmrossy) [reply src/features/tokens/metadata.ts:6]: The CLI artifact file name is `warp-ui-token-config.json` but I don't recommend we use that long name here

**Pull request merged**